nginx_stage/lib/nginx_stage/user.rb

```diff
@@ -33,21 +33,33 @@ class User
     # @param user [String] the user name defining this object
     # @raise [ArgumentError] if user or primary group doesn't exist on local system
     def initialize(user)
-      @passwd = Etc.getpwnam user.to_s
-      @group = Etc.getgrgid gid
-      @groups = get_groups
-
-      if name.to_s != user.to_s
-        err_msg = <<~HEREDOC
-          Username '#{user}' is being mapped to '#{name}' in SSSD and they don't match.
-          Users with domain names cannot be mapped correctly. If '#{name}' still has the
-          domain in it you'll need to set SSSD's full_name_format to '%1$s'.
+      # See if user is all numbers (potentially uid), regexp is 8% faster than integer conversion
+      # Benchmark: 0.13 microseconds per call (1M cycles)
+      if user.match?(/\A\d+\z/)
+        # The user is composed of all numbers, (numeric string)
+        #  10 microseconds per call if it matches
+        # 203 microseconds per call if it doesn't match (only happens if username is all numbers but not a uid)
+        @passwd = Etc.getpwuid(user.to_i) rescue nil
+      end
 
-          See https://github.com/OSC/ondemand/issues/1759 for more details.
-        HEREDOC
+      # Variable is not set, so the user is a string
+      if not @passwd
```

Comment:
This is a little bit of a nit-pick, but idiomatic ruby here is `unless` instead of `if not`.